### PR TITLE
core-1354 Allow users to rate apps from notification message.

### PR DESCRIPTION
**Pull request by sri2k1us**
- Rating widget
image

- User Rating
image

- iPhone 5
image

### CODE REVIEW

File: src/components/apps/RatingWidget.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        });
+    };
+
+    useEffect(() => {
+        if (selectedApp) {
+            setAppRating(selectedApp.rating);
+        }
+    }, [selectedApp]);
+
+    const loading = appByIdStatus || ratingMutationStatus === constants.LOADING;
```
Comment (ash3rz) [inline src/components/apps/RatingWidget.js:75]: I think this was meant to check both variables for the loading status:
```suggestion
    const loading = appByIdStatus === constants.LOADING || ratingMutationStatus === constants.LOADING;
```

This should also work:
```
import isQueryLoading from "components/utils/isQueryLoading";

const loading = isQueryLoading([appByIdStatus, ratingMutationStatus])
```

**Review by ash3rz — APPROVED** [review]
LGTM :shipit: 
One minor thing that I think would need to come from ui-lib, is the trash can is a little misaligned with the rating stars when they're on the same line.

File: src/components/apps/RatingWidget.js (hunk 1 above)
Comment (psarando) [reply src/components/apps/RatingWidget.js:75]: The `appByIdStatus` var is actually an alias for the app listing query's `isFetching` flag, so this works, though maybe the alias is misnamed. The `isQueryLoading` util function is still a good idea 👍

File: src/components/notifications/Message.js [hunk 2]
```diff
@@ -65,7 +67,25 @@ function AnalysisLink(props) {
...
+                    notification.payload?.status === analysisStatus.FAILED ||
+                    notification.payload?.status === analysisStatus.CANCELED);
+            return (
+                <>
+                    <MessageLink message={message} href={href} as={as} />
+                    <br />
+                    {allowRating && (
+                        <RatingWidget
+                            appId={notification.payload?.app_id}
+                            appName={notification?.payload.app_name}
```
Comment (psarando) [inline src/components/notifications/Message.js:83]: Other places in this code are checking for `notification.payload?.`, but here it's `notification?.payload.`

Maybe these could be refactored into a `payload` var that replaces all of them?
```javascript
const payload = notification?.payload;
```
Then `payload?.app_name` etc.

**Review by psarando — APPROVED** [review]
LGTM 👍 
Though I had a couple of minor refactor suggestions.

**Comment by sri2k1us** [comment]
> LGTM :shipit:
> One minor thing that I think would need to come from ui-lib, is the trash can is a little misaligned with the rating stars when they're on the same line.

Will file an issue for this and move-on since there are other priority items to work on now.

**Comment by sri2k1us** [comment]
Thanks for the reviews.

**Pull request merged**